server/plugin/api.go

```diff
@@ -363,41 +419,60 @@ func (p *Plugin) completeConnectUserToGitHub(c *Context, w http.ResponseWriter,
 	}
 
 	if err = p.storeGitHubUserInfo(userInfo); err != nil {
-		p.API.LogWarn("Failed to store GitHub user info", "error", err.Error())
-		http.Error(w, "Unable to connect user to GitHub", http.StatusInternalServerError)
+		c.Log.WithError(err).Warnf("Failed to store GitHub user info")
+
+		rErr = errors.Wrap(err, "Unable to connect user to GitHub")
+		http.Error(w, rErr.Error(), http.StatusInternalServerError)
 		return
 	}
 
 	if err = p.storeGitHubToUserIDMapping(gitUser.GetLogin(), state.UserID); err != nil {
-		p.API.LogWarn("Failed to store GitHub user info mapping", "error", err.Error())
+		c.Log.WithError(err).Warnf("Failed to store GitHub user info mapping")
 	}
 
-	commandHelp, err := renderTemplate("helpText", p.getConfiguration())
+	flow := p.flowManager.setupFlow.ForUser(c.UserID)
+
+	stepName, err := flow.GetCurrentStep()
 	if err != nil {
-		p.API.LogWarn("Failed to render help template", "error", err.Error())
-	}
-
-	// Post intro post
-	message := fmt.Sprintf("#### Welcome to the Mattermost GitHub Plugin!\n"+
-		"You've connected your Mattermost account to [%s](%s) on GitHub. Read about the features of this plugin below:\n\n"+
-		"##### Daily Reminders\n"+
-		"The first time you log in each day, you will get a post right here letting you know what messages you need to read and what pull requests are awaiting your review.\n"+
-		"Turn off reminders with `/github settings reminders off`.\n\n"+
-		"##### Notifications\n"+
-		"When someone mentions you, requests your review, comments on or modifies one of your pull requests/issues, or assigns you, you'll get a post here about it.\n"+
-		"Turn off notifications with `/github settings notifications off`.\n\n"+
-		"##### Sidebar Buttons\n"+
-		"Check out the buttons in the left-hand sidebar of Mattermost.\n"+
-		"* The first button tells you how many pull requests you have submitted.\n"+
-		"* The second shows the number of PR that are awaiting your review.\n"+
-		"* The third shows the number of PR and issues your are assiged to.\n"+
-		"* The fourth tracks the number of unread messages you have.\n"+
-		"* The fifth will refresh the numbers.\n\n"+
-		"Click on them!\n\n"+
-		"##### Slash Commands\n"+
-		commandHelp, gitUser.GetLogin(), gitUser.GetHTMLURL())
-
-	p.CreateBotDMPost(state.UserID, message, "custom_git_welcome")
+		c.Log.WithError(err).Warnf("Failed to get current step")
+	}
+
```

Comment:
What happens when this user is not an admin? Do we want to log this warning unconditionally here?

Reply:
`GetCurrentStep()` should return `"", nil` where is no flow running. At least this is show the library used to work. @levb Would you mind changing the method in your PR to behave that way?

Reply:
https://github.com/mattermost/mattermost-plugin-api/issues/113